parsl/app/futures.py

```diff
@@ -31,15 +34,26 @@ def parent_callback(self, parent_fu):
         Returns:
             - None
         """
-
         e = parent_fu._exception
         if e:
             self.set_exception(e)
         else:
             self.set_result(self.file_obj)
+            if self.file_obj.scheme == 'file' and os.path.isfile(self.file_obj.filepath):
+                if not self.file_obj.timestamp:
+                    self.file_obj.timestamp = datetime.fromtimestamp(stat(self.file_obj.filepath).st_ctime, tz=timezone.utc)
+                if not self.file_obj.size:
+                    self.file_obj.size = stat(self.file_obj.filepath).st_size
+                if not self.file_obj.md5sum:
+                    self.file_obj.md5sum = md5(open(self.file_obj, 'rb').read()).hexdigest()
```

Comment:
md5 is quite an expensive operation

I think this will happen on every file even if file provenance has not been enabled by the user?

If so, that's a bad thing.

Reply:
Yes, as written it would happen every time, which I agree is not good. I will rework the code to only trigger this when provenance tracking is on.